### PR TITLE
reduce size of dask-notebook Docker image by 0.25GB

### PR DESCRIPTION
**Pull request by jeffreybreen**
Hi:

I only just noticed #47, but I think my changes are orthogonal. I reduced the size of the dask-notebook image by ~250MB:

1. Chained the `conda` commands into the same layer; same for `pip`
2. Bumped the version of jupyterlab to 0.35.5 to match what's already included in the `jupyter/base-notebook` base image thereby eliminating a downgrade. To make this stick, though, I also specified the current tag of the base image -- not sure how you guys feel about that...
3. Lowered the version of `jupyter-widgets/jupyterlab-manager` to 0.38 to avoid downloading the default 0.40 which wouldn't install

```
$ docker images | grep 'daskdev/dask-notebook'
daskdev/dask-notebook                           after                          5f5847bac1f6        30 minutes ago      1.5GB
daskdev/dask-notebook                           latest                         507f657f6fab        2 days ago          1.75GB
daskdev/dask-notebook                           1.2.0                          f5c7ffbc4615        3 weeks ago         1.7GB
```

In addition to the potential changes of #47, there still seems to be a lot of noise from `npm` during the installation of `dask-labextension`. I'm no Node expert, but eliminating these multiple versions should be low-hanging fruit for someone who is:

```
WARNING in d3-array
  Multiple versions of d3-array found:
    1.2.4 ./~/d3-scale/~/d3-array from ./~/d3-scale/~/d3-array/src/index.js
    2.0.3 ./~/d3-array from ./~/d3-array/src/index.js

WARNING in react
  Multiple versions of react found:
    16.4.2 ./~/react from ./~/react/index.js
    16.8.6 ./~/dask-labextension/~/react from ./~/dask-labextension/lib/dashboard.js

WARNING in react-dom
  Multiple versions of react-dom found:
    16.4.2 ./~/react-dom from ./~/react-dom/index.js
    16.8.6 ./~/dask-labextension/~/react-dom from ./~/dask-labextension/~/react-dom/index.js

WARNING in vega-lite
  Multiple versions of vega-lite found:
    2.5.1 ./~/vega-lite/build/src from ./~/vega-lite/build/src/compile/selection/selection.js
    2.6.0 ./~/vega-lite/build from ./~/vega-lite/build/src/index.js
```

### CODE REVIEW

**Comment by mrocklin** [comment]
Thanks @jeffreybreen .  Reducing image size is definitely something that positively impacts the Dask community.  

cc'ing @quasiben @jakirkham here for review, if they have time.

Also cc @jhamman in case he's interested in similar changes for Pangeo

File: notebook/Dockerfile [hunk 1]
```diff
@@ -1,37 +1,45 @@
-FROM jupyter/base-notebook
+FROM jupyter/base-notebook:4d7dd95017ed
```
Comment (jakirkham) [inline notebook/Dockerfile]: Why this specific hash?

**Comment by jakirkham** [comment]
One item would be good to discuss. Otherwise this generally seems fine.

Though @jcrist is working on some other size reductions in PR ( https://github.com/dask/dask-docker/pull/47 ). IDK how we want to handle that.

File: notebook/Dockerfile (hunk 1 above)
Comment (jeffreybreen) [reply notebook/Dockerfile]: That was "latest" of the base image when I made my edits. If we don't specify a version, the base image may have newer versions than we're requesting, triggering a downgrade in some packages. 

This happened recently and caused some bloat in our image: the current base image contains `jupyterlab==0.35.5` but we were calling for `jupyterlab==0.35.4`, so it downloaded and installed the lower version.

Another way around this is would be less restrictive in specifying package versions: `jupyterlab>=0.35.4` should also avoid the downgrade.

**Comment by jcrist** [comment]
Thanks @jeffreybreen - I'm going to merge this for now, but there will likely be some changes again after #47.

**Pull request merged**